plugins/ledger/ledger.py

```diff
@@ -321,47 +315,38 @@ def sign_transaction(self, tx, password):
             # Get trusted inputs from the original transactions
             for utxo in inputs:
                 sequence = int_to_hex(utxo[5], 4)
-                if not p2shTransaction:
-                    txtmp = bitcoinTransaction(bfh(utxo[0]))
-                    trustedInput = self.get_client().getTrustedInput(txtmp, utxo[1])
-                    trustedInput['sequence'] = sequence
-                    chipInputs.append(trustedInput)
-                    redeemScripts.append(txtmp.outputs[utxo[1]].script)
-                else:
-                    tmp = bfh(utxo[3])[::-1]
-                    tmp += bfh(int_to_hex(utxo[1], 4))
-                    chipInputs.append({'value' : tmp, 'sequence' : sequence})
-                    redeemScripts.append(bfh(utxo[2]))
+                txtmp = bitcoinTransaction(bfh(utxo[0]))
+                tmp = bfh(utxo[3])[::-1]
+                tmp += bfh(int_to_hex(utxo[1], 4))
+                tmp += txtmp.outputs[utxo[1]].amount
+                chipInputs.append({'value' : tmp, 'witness' : True, 'sequence' : sequence})
+                redeemScripts.append(bfh(utxo[2]))
 
             # Sign all inputs
-            firstTransaction = True
             inputIndex = 0
             rawTx = tx.serialize()
             self.get_client().enableAlternate2fa(False)
-            if True:
-                while inputIndex < len(inputs):
-                    self.get_client().startUntrustedTransaction(firstTransaction, inputIndex,
-                                                            chipInputs, redeemScripts[inputIndex])
-                    outputData = self.get_client().finalizeInputFull(txOutput)
-                    outputData['outputData'] = txOutput
-                    if firstTransaction:
-                        transactionOutput = outputData['outputData']
-                    if outputData['confirmationNeeded']:
-                        outputData['address'] = output
-                        self.handler.finished()
-                        pin = self.handler.get_auth( outputData ) # does the authenticate dialog and returns pin
-                        if not pin:
-                            raise UserWarning()
-                        if pin != 'paired':
-                            self.handler.show_message(_("Confirmed. Signing Transaction..."))
-                    else:
-                        # Sign input with the provided PIN
-                        inputSignature = self.get_client().untrustedHashSign(inputsPaths[inputIndex], pin, lockTime=tx.locktime)
-                        inputSignature[0] = 0x30 # force for 1.4.9+
-                        signatures.append(inputSignature)
-                        inputIndex = inputIndex + 1
-                    if pin != 'paired':
-                        firstTransaction = False
+            self.get_client().startUntrustedTransaction(True, inputIndex,
```

Comment:
nice clean up.

Reply:
Only [BIP143 signing](https://github.com/spesmilo/electrum/blob/master/plugins/ledger/ledger.py#L378) is needed